core/src/main/java/edu/wpi/grip/core/metrics/FixedSizeQueue.java

```diff
@@ -0,0 +1,115 @@
+package edu.wpi.grip.core.metrics;
+
+import java.util.AbstractQueue;
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * A queue implementation that, once full, removes the oldest element before adding a new one.
+ */
+public class FixedSizeQueue<E> extends AbstractQueue<E> {
```

Comment:
Cool! I like it!
However, google beat you to it on this one:
https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/EvictingQueue.java